packages/ui-components/src/components/Badge/Badge.test.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as React from "react"
+import { describe, expect, test } from "vitest"
+import { render, screen } from "@testing-library/react"
+
+import { KnownIcons } from "../Icon/Icon.component"
+
+import { VariantType } from "./Badge.types"
+import { Badge } from "./"
+
+describe("Badge component", () => {
+  // BASIC RENDERING
+
+  test("renders with default props", () => {
+    render(<Badge data-testid="badge" />)
+    expect(screen.getByTestId("badge")).toBeInTheDocument()
+    expect(screen.getByTestId("badge")).toHaveClass("juno-badge-default")
+    expect(screen.queryByRole("img")).not.toBeInTheDocument()
+  })
+
+  test("renders a badge with text", () => {
+    render(<Badge text="default badge" data-testid="badge" />)
+    expect(screen.getByTestId("badge")).toBeInTheDocument()
+    expect(screen.getByTestId("badge")).toHaveTextContent("default badge")
+  })
+
+  test("renders a badge with children", () => {
+    render(<Badge data-testid="badge">Children inside</Badge>)
+    expect(screen.getByTestId("badge")).toBeInTheDocument()
+    expect(screen.getByTestId("badge")).toHaveTextContent("Children inside")
+  })
+
+  // VARIANTS
+
+  const variants: VariantType[] = ["info", "success", "warning", "danger", "error", "critical"]
+  variants.forEach((variant) => {
```

Comment:
I would suggest to not reuse the tests here with forEach, duplications in tests are even suggestible so lets discuss it too.